### ISSUE TITLE
Feature request to be able to exclude websites

### ISSUE THREAD

**Issue by o-be-one** [ClearURLs/Addon#93]
Hello,

i wanted to upload a file on Tresorit website's and i was not able because ClearURL was cleaning some part of it. I had few 412 HTTP status. I haven't digged deep but i've seen that disabling this extension (latest stable Firefox in my case) resolved my issue.
So, i've searched to add Tresorit in an exclusion list as i can do with few other addons but haven't seen that.

Is it possible to add a feature to disable ClearURLs options for selected websites? Or at least can someone fix the file upload blocking we experience on Tresorit website?

Thanks a lot for your help.

**Comment by deadbeef2000** [comment]
Hello,

same problem occurs when I'm trying to visit the following page:

https://app.getresponse.com/click.html?x=a62b&lc=SX7CKv&mc=In&s=NBb4KD&u=zs9vf&z=ES3yhIs&

The filtered version doesn't include `mc=ln` and the trailing &, if this part is removed, I get redirected to a 404 page, but if I disable the addon, I get redirected properly.

This happens to **all** app.getresponse.com/click.html links, but unsubscribe.html or change.html works perfectly fine.

**Comment by offobject** [comment]
I would like this feature as well. It seems to break these embedded Vimeo videos:
https://university.vectorworks.net/mod/scorm/player.php?a=12&currentorg=articulate_rise&scoid=24

Opening the frame in another (non-embedded) window does not fix. On latest stable Firefox.

**Comment by Ugo-Gira** [comment]
I agree, please add a whitelisting feature or a button to enable/disable the add-on on each website. Not to sound ungrateful, but, seriously, it's like the most basic feature in something which has the potential of breaking things.

**Comment by flo-dt** [comment]
I also need a whitelist feature. I leave the addon uninstalled until this feature has been added.

**Comment by Ryonez** [comment]
Have also had instances that I need to allow it of pages won't load for a service.

When this occurs I typically disable the extension and never turn it back on for ages because I'll forget about it.

**Comment by kanish671** [comment]
The `mc` query param in this link https://www.mailing.dzone.com/click.html?x=a62e&lc=toz&mc=h&s=TSmS&u=f&z=oTUWUqe& is getting removed and causing to redirect to 404 page. I would like to have an option to whitelist certain websites or certain params in URLs of certain websites.

**Comment by oleghristov** [comment]
Also throwing my hat in the ring for this feature request

**Comment by securingmom** [comment]
This seems like it would be better if it had a list subscription option so all y'all could help build a peer reviewed whitelist for optional implementation.

**Comment by LM1LC3N7** [comment]
As the extension manipulate URLs, this can led to some false positives.
So adding an exclusion option, like uBlock (you can disable the extension for the current visited domain) would be a great  idea 😄

**Comment by Shamp0o** [comment]
+1
This add-on makes my University's websites unusable.

**Comment by VictorioBerra** [comment]
Please add this.

**Comment by brendandburns** [comment]
fwiw, creating new GitHub CodeSpaces is also broken by this.

**Comment by Ansem93** [comment]
This should be top priority. I have important websites not working because of clearurls.

**Comment by DaBomby** [comment]
This is a huge deal breaker in terms of using this addon. A simple all rules are on or all rules are off functionality is not helpful. It turns into a situation wherein you will have to turn it off to make sites work, and then you forget to turn it back on. Ultimately eroding the value proposition from this extension. 

This should be made a priority for a new feature release.

**Comment by Ryonez** [comment]
@KevinRoebert, lucky you for the mention. No-one working on this has responded here and you have the most recent commits.

To give you an idea how bad this is, this breaks websites. Legitimate ones some of us need to use.
Where the aim is to improve browsing experience and maintain users privacy, the inability to use legitimate services that we need to does the opposite, resulting in the extension being disabled or removed.

This should be considered a key required feature for the extension.

**Comment by DaBomby** [comment]
> @KevinRoebert, lucky you for the mention. No-one working on this has responded here and you have the most recent commits.
> This should be considered a key required feature for the extension.

Agree, there does not seem to be any response and the updates to the code are rather infrequent. I wonder if this project has become abandoned? 

@KevinRoebert or @yipinghuang1991 are you able to weigh in on this?

**Comment by ahuj9** [comment]
Can we just add a "disable on this site" button to the extension?

**Comment by Caligatio** [comment]
This add-on now breaks the new login flow for the US's third largest credit union. It would be wonderful if we could get some ability to whitelist sites, even if it isn't he prettiest implementation.

**Comment by GuardianLiarus** [comment]
Yes please, we need this feature, some sites break due to clearurls for me

**Comment by learn2reid** [comment]
Is there any update on this?

**Comment by thakyZ** [comment]
I would like this mainly because I believe filtering links on Steam's link filter wouldn't be a good idea and might be a security risk. So I would like to piece of mind to not have that be bypassed.

**Comment by qmzmdqs** [comment]
I also need this feature.
I hope it can be temporarily disabled at the specified host and URL. Because some rules have errors, which affect the normal function of the webpage

**Comment by diogoleitao** [comment]
Chiming in on this as well, HERE WeGo breaks a lot with this extension as well.

**Comment by abudhu** [comment]
Also adding my desire for this feature.  Please!

**Comment by pjboro** [comment]
This addon is breaking some popular websites. It would be awesome to be able to turn it off for specific websites.
Would devs accept a PR, provided someone created one?

**Comment by VictorioBerra** [comment]
Someone just needs to fork this, or create a PR and pray [KevinRoebert](https://github.com/KevinRoebert) accepts it.

Last month he closed and commented on a PR.

**Comment by NoFrans** [comment]
Hopefully someone can implement this. 🤞

**Comment by gbakeman** [comment]
For everyone's info, [there's been some discussion](https://www.reddit.com/r/PrivacyGuides/comments/rfzhzl/actually_legitimate_url_shortener_tool_vs/)  about whether or not CleanURLs is even necessary with the appropriate uBlock lists. It sounds like you can replace the functionality of CleanURLs with `AdGuard URL Tracking Protection` and `Actually Legitimate URL Shortener Tool`.

**Comment by Caligatio** [comment]
To shortccut @gbakeman's reddit links:

- Enable **AdGuard URL Tracking Protection** (present by default)
- Import **[➗ Actually Legitimate URL Shortener Tool](https://raw.githubusercontent.com/DandelionSprout/adfilt/master/LegitimateURLShortener.txt)**
- Import **[ClearURLs for uBo](https://raw.githubusercontent.com/DandelionSprout/adfilt/master/ClearURLs%20for%20uBo/clear_urls_uboified.txt)** (this is by the same project as ➗ Actually Legitimate URL Shortener Tool)

The biggest upside to using the uBo functionality is that you can use these lists on mobile whereas ClearURLs isn't supported.

**Comment by posita** [comment]
See also https://github.com/ClearURLs/Addon/issues/144#issuecomment-1002811422.

**Comment by keldian** [comment]
> For everyone's info, [there's been some discussion](https://www.reddit.com/r/PrivacyGuides/comments/rfzhzl/actually_legitimate_url_shortener_tool_vs/) about whether or not CleanURLs is even necessary with the appropriate uBlock lists. It sounds like you can replace the functionality of CleanURLs with `AdGuard URL Tracking Protection` and `Actually Legitimate URL Shortener Tool`.

uBO has no support for bypassing redirect tracker URLs, never mind how many lists you add to it. That specific mechanism is the entire reason I started using ClearURLs.

**Comment by IDUnavailable** [comment]
[The author said he'd be working on it more after Feb '22 and wanted to put in the whitelist feature](https://github.com/ClearURLs/Addon/issues/144#issuecomment-1002811422), though it seems @KevinRoebert hasn't posted on this issue about it.

I'm using [Tautulli](https://github.com/Tautulli/Tautulli) and "Login with Plex" no longer works. ClearURLs mangles the URL so that it just points at the frontpage of the Plex website. I tried fiddling with ClearURL's options but it seems none of them are specifically causing it. I was only able to fix it by disabling ClearURLs altogether.

It would be infinitely easier to have a whitelist feature where I could fix this in seconds, as opposed to having to disable the extension and write up an issue here for every single possible website that this could happen to.

**Comment by m4jki** [comment]
broken captcha login on gate.io with filter enabled

**Comment by xenstar** [comment]
wow, so many people have requested this. Now if I disable it for a specific site, I have to disable it for all the sites. And sometimes I even forget to enable it. This is a good plugin, I hope the developer will add a whitelist function soon.

![image](https://user-images.githubusercontent.com/55502651/186067824-b5fd8618-749b-4853-8093-c18a26c4a628.png)

**Comment by gbakeman** [comment]
As mentioned earlier, you can supplement most of CleanURL's functionality with uBlock lists.

https://github.com/ClearURLs/Addon/issues/93#issuecomment-1111052855

**Comment by Brog33** [comment]
love this plugin but it breaks certain sites, so i will disable it until a whitelist is in place.

**Comment by learn2reid** [comment]
+1!

**Comment by Drugoy** [comment]
@EsEnZeT <a href="https://chromewebstore.google.com/detail/neat-url/jchobbjgibcahbheicfocecmhocglkco">NeatURL</a> - a better alternative that has the blacklist functionality (which this ticket is all about) and whose UI I like more (instead of showing me bullshit like statistics - get me direct access to the GET param cleanup rules!).

**Comment by OutlawCecil** [comment]
Agreed this is needed! AliExpress is broken by this extension currently.

**Comment by nxmndr** [comment]
I have tried editing the data.minify.json & rules.minify.hash URLs with new ones giving custom JSON but they are not saved clicking Save & restart.

**Comment by Banaanae** [comment]
I've started working on adding a whitelist, I'll make a draft PR once basic functionality is added

**Comment by nxmndr** [comment]
Thanks :) Ideally it should be per parameter and per website, as some legit website also have some parameter mishmash.

**Comment by Banaanae** [comment]
I'll see what I can do, but that might be a bit advanced for me :)

Also most whitelist that come to mine (uBo sponsorblock) don't even have this functionality

However when I submit my PR anyone could implement that

**Comment by Banaanae** [comment]
This has been completed in #372

**Comment by Acleacius** [comment]
New issue at least to me, when using Google Maps it keeps the url locked so Google can't zoom in and auto readjust
positioning. It also means many normal functions on Google Maps don't work such as clicking on a POI to get the popup 
 and all the information provided. Such as Hours of Operation, Crowd Size (haha) atm and Addresses.

Personally, I still use this mod BUT it's really good to get a list of places and situations, so we can be aware to disable manually
as necessary.

**Comment by Speedy2662** [comment]
This extension breaks Google Docs / Sheets. I'd love to disable the extension just for that domain, please implement this feature...

**Comment by kult** [comment]
Great application. I'm hoping for this feature too! 
I want to whitelist Aliexpress domains and a few others that get totally hosed when CleanURLs is active.

**Comment by BlaineEXE** [comment]
Ditto. ClearURLs breaks Aetna login page also.

**Comment by UnconnectedBedna** [comment]
Yeah, I have been using this tool for years and today it broke a search function on a site. Took me a while to figure out it was clearURLs that caused the issue.

If exceptions can not be configured, this tool has to go for me...

**Comment by vgeorgiev69** [comment]
Not sure how, but I assume some recent updates broke something in the addon. Aliexpress pages suddenly (from yesterday at least) started showing 404 to every new search I make. It opened previous items that i've opened, but not a single new one.
It redirected me to various languages, but all the pages were 404 not found.

Disabling the extension fixes the problem.

**Comment by wzrayyy** [comment]
+1, Google Sheets somehow uses GET parameters in the URL to open pop-up menus and breaks with the add-on enabled.

**Comment by HiGein** [comment]
Have the same issue with sheets.

**Comment by bigolbooty** [comment]
Just +1ing this ask. I can no longer use my bank with this extension active, I will have to stop using for the time being until a whitelist is added 🫡

**Comment by Speedy2662** [comment]
I find it unbelievable that the dev is ignoring this simple request since 2021. The extension is actively promoted by Firefox, and there seems to be no desire to actually improve upon the extension?

**Comment by Ryonez** [comment]
Yeah, I'm unsubbing from this. Left the extension long ago, and there's been no improvements on the issue.

**Comment by UnconnectedBedna** [comment]
#372 #353 
https://gitlab.com/ClearURLs/ClearUrls/-/merge_requests/110

```
[Kevin](https://gitlab.com/KevinRoebert) added 7 commits [8 months ago](https://gitlab.com/ClearURLs/ClearUrls/-/merge_requests/110#note_1998950271)

    [deec80b7](https://gitlab.com/ClearURLs/ClearUrls/-/commit/deec80b763179fa5c3559a37e3c9a6f1b28d0886) - 1 commit from branch master
    [e86654ed](https://gitlab.com/ClearURLs/ClearUrls/-/merge_requests/110/diffs?commit_id=e86654ed29888d82b029972eac3827529c5afaca) - first commit for adding whitelist
    [6a638596](https://gitlab.com/ClearURLs/ClearUrls/-/merge_requests/110/diffs?commit_id=6a63859635099345422bb0588778d980186f9a13) - allowing editing of whitelisted sites in settings
    [da90e259](https://gitlab.com/ClearURLs/ClearUrls/-/merge_requests/110/diffs?commit_id=da90e259bb14564aadc22e4ee54cc6ddb905b067) - handle url better; fix saving when edited
    [81eb931e](https://gitlab.com/ClearURLs/ClearUrls/-/merge_requests/110/diffs?commit_id=81eb931e0262adbb4df97fd074f232fd6db93937) - fix styling; allow removing of sites
    [fcbe2abf](https://gitlab.com/ClearURLs/ClearUrls/-/merge_requests/110/diffs?commit_id=fcbe2abfdd65edcefff9a4ab22bddc7e3f645eb3) - Fix updating whitelist button
    [08de228c](https://gitlab.com/ClearURLs/ClearUrls/-/merge_requests/110/diffs?commit_id=08de228cc557536035693bbe0aa8c3f42e5621bb) - changes
```

Any news?
I kept the plugin, but as of today when fingerprinting is how you are tracked, more plugins makes it easier to fingerprint you.
So if this feature does not get implemented, it makes little sense to keep this plugin.

Could we at least get an acknowledgement from dev that it IS being worked on?

**Comment by sreid99** [comment]
Also stopped using this extension due to this issue some time ago. Now unfollowing this thread as doubt it will ever be fixed.

**Comment by lxvi** [comment]
Hi, this issue has been open for more than four years, and is obviously very popular.  Please indicate where on the roadmap it will be addressed.

**Comment by wolfie** [comment]
This also breaks [OnShape](https://www.onshape.com/en/)'s search function

**Comment by onlinemumie** [comment]
My bank started to block access because of this. Since there's no whitelist, I must either remember to manually deactivate at every login, or deactivate the add-on. For the time being, I will deactivate the login.

**Comment by o-data** [comment]
You guys think with AI we can get it to add this feature?

**Comment by Banaanae** [comment]
No point, if human written code doesn't get merged I'm doubtful that ai written stuff would.

I recommend using https://github.com/DandelionSprout/adfilt/blob/master/LegitimateURLShortener.txt
More tracking is removed and they respond faster to breakage, doesn't have some features like redirects but I think it's worth it to not have every second site broken.

**Comment by nisc** [comment]
This is desperately needed. Had to remove ClearURLs for now, given it breaks too many sites. It often took me months to realize that ClearURLs was causing a particular issue.

**Comment by shalak** [comment]
Same thing with me. I cannot order pizza, because this request:

```
https://www.pizzeriaeltoro.pl/dodaj.php?typ=PIZZA&gid=2&pid=10&ref=-1&r=1
```
becomes this:

```
https://www.pizzeriaeltoro.pl/dodaj.php?typ=PIZZA&gid=2&pid=10&r=1
```
and the backend breaks because of missing ref param :(
